components/raftstore-v2/src/router/message.rs

```diff
@@ -157,7 +157,7 @@ pub enum PeerMsg {
     /// Raft message is the message sent between raft nodes in the same
     /// raft group. Messages need to be redirected to raftstore if target
     /// peer doesn't exist.
-    RaftMessage(Box<RaftMessage>),
+    RaftMessage(Box<RaftMessage>, Option<Instant>),
```

Comment:
How about use a object to encapsulate the metrics?
Such as:
```Rust
struct Metrics {
    send_time: Instant,
    /// ... add more timestamp.
}
```
Or just like the implementation in `raftstore-v1` -- `InspectRaftMessage` does.
https://github.com/tikv/tikv/blob/b4e0bf7bab6ad395b74c0be938119d82ded4cd2a/components/raftstore/src/store/msg.rs#L733-L736
Add one extra `send_time: Instant` into it?

Reply:
See the comments below.